app/src/firebase/snaps/SnapService.ts

```diff
@@ -0,0 +1,32 @@
+import Snap from "./Snap";
+import firebase from "firebase/app";
+import "firebase/auth";
+
+export async function submitSnap(snap: Snap) {
+    await firebase.firestore().collection("snaps").add(snap);
+}
+
+function docToSnap(data: any) {
+    return {
+        body: data.body,
+        timestamp: data.timestamp.toDate(),
+        to: data.to,
+        from: data.from,
+    };
+}
+
+async function getSubmittedSnaps(user: firebase.User): Promise<Snap[]> {
+    // .orderBy("timestamp", "desc")
```

Comment:
This isn't a styling thing. What's blocking this feature?

Reply:
Firestore wants you to set up and save some kind of custom query on your database if you want to chain filters / sorts in order to improve performance. I don’t mind to do that.